clients/geth/specular/rollup/services/base.go

```diff
@@ -65,6 +61,26 @@ func (b *BaseService) Chain() *core.BlockChain {
 	return b.Eth.BlockChain()
 }
 
+
+func (b *BaseService) GetL1RollupBlock(ctx context.Context, start uint64) uint64 {
+	l1blocknumber := start
+	l1BlockHead, err := b.L1Client.BlockNumber(ctx)
+	if err != nil {
+		return b.Config.L1RollupGenesisBlock
+	}
+	opts := bind.FilterOpts{Start: b.Config.L1RollupGenesisBlock, End: &l1BlockHead, Context: ctx}
+	eventsIter, err := b.L1Client.FilterTxBatchAppendedEvents(&opts)
+	if err != nil {
+		return b.Config.L1RollupGenesisBlock
+	}
+	for eventsIter.Next() {
+		if eventsIter.Event.Raw.BlockNumber > l1blocknumber{
```

Comment:
i don't really understand what the intent is here, you're just getting the latest tx batch's starting block number; that doesn't tell you anything about whether it corresponds to the local state

also please apply go format (can you enable this in your IDE? its automatic in vscode)